tests/components/moehlenhoff_alpha2/test_config_flow.py

```diff
@@ -0,0 +1,91 @@
+"""Test the moehlenhoff_alpha2 config flow."""
+import asyncio
+from unittest.mock import patch
+
+from homeassistant.components.moehlenhoff_alpha2.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+
+async def mock_fetch_static_data(self):
+    """Mock moehlenhoff_alpha2.Alpha2Base._fetch_static_data."""
+    self.static_data = {
+        "Devices": {"Device": {"ID": "fake_id", "NAME": "fake_base_name"}}
+    }
+
+
+async def test_duplicate_error(hass: HomeAssistant):
+    """Test that errors are shown when duplicates are added."""
+    with patch(
+        "moehlenhoff_alpha2.Alpha2Base._fetch_static_data", mock_fetch_static_data
+    ):
+        result = await hass.config_entries.flow.async_init(
```

Comment:
Add a MockConfigEntry instead with the same host as entered in the flow.

https://github.com/home-assistant/core/blob/76bfbbafe1ef3761c75e397c285e8057db926fe4/tests/common.py#L755

Reply:
I have updated the tests based on the scaffold template